### PR TITLE
Enterprise install and authentication configuration doc

### PR DESCRIPTION
**Pull request by jsirianni**
### Proposed Change

These changes will make it to the doc site once merged.

- Added quick start section for installing enterprise
- Added config doc for enterprise authentication

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -412,3 +412,164 @@ tls_config:
...
+| `authType`   | Authentication type to use (`ldap` / `active-directory` ). |
+| `ldapConfig` | Contains configuration for ldap and active directory. |
+
+**LDAP / Active Directory Configuration**
+
+The configuration's `server.ldapConfig` secttion contains the following options for
+configuration  LDAP and Active Directory.
+
+| Option              | Description |
+| ------------------- | ----------- |
```
[the remark this answers is not in the document]
Comment (Mrod1598) [reply docs/configuration.md:438]: description could be some thing like 
```
the default protocol used for connections is `ldap` to make a tls enabled connection set protocol to `ldaps`
```

File: docs/configuration.md [hunk 2]
```diff
@@ -412,3 +412,164 @@ tls_config:
...
+password: medora5234
+logFilePath: /var/log/bindplane/bindplane.log
+server:
+    storageFilePath: /var/lib/bindplane/storage/bindplane.db
+    secretKey: e8bfcfe0-bbe6-4ee6-bf35-72ff182d2dc5
+    remoteURL: ws://bindplane.corp.net:3001
+    sessionsSecret: b84746f6-aca1-4b91-9ccd-d4da8d75fe4d
+    authType: ldap
+    ldapConfig:
+        protocol: ldap
```
Comment (Mrod1598) [inline docs/configuration.md:563]: protocol should be `ldaps` for tls

**Review by antonblock — APPROVED** [review]
Really great writeup for the LDAP auth

File: docs/configuration.md [hunk 3]
```diff
@@ -412,3 +412,164 @@ tls_config:
...
+        server: ldap.corp.net
+        port: "389"
```
Comment (Mrod1598) [inline docs/configuration.md:534]:
```suggestion
        port: "636"
```

File: docs/configuration.md [hunk 4]
```diff
@@ -412,3 +412,164 @@ tls_config:
...
+password: medora5234
+logFilePath: /var/log/bindplane/bindplane.log
+server:
+    storageFilePath: /var/lib/bindplane/storage/bindplane.db
+    secretKey: e8bfcfe0-bbe6-4ee6-bf35-72ff182d2dc5
+    remoteURL: ws://bindplane.corp.net:3001
+    sessionsSecret: b84746f6-aca1-4b91-9ccd-d4da8d75fe4d
+    authType: ldap
+    ldapConfig:
+        protocol: ldap
```
Comment (Mrod1598) [inline docs/configuration.md:532]:
```suggestion
        protocol: ldaps
```

**Review by ryancgoins — APPROVED** [review]
This looks great Joe

File: docs/configuration.md [hunk 5]
```diff
@@ -412,3 +412,164 @@ tls_config:
...
+
+This example is the same as the "basic" example, with TLS. The protocol has been set to `ldaps`, port to `636`, and a ca certificate is optionally* configured.
+
+```yaml
+host: 0.0.0.0
+port: "3001"
+serverURL: http://bindplane.corp.net:3001
+username: medora
+password: medora5234
+logFilePath: /var/log/bindplane/bindplane.log
```
Comment (Mrod1598) [inline docs/configuration.md:524]: Are we giving this example specifically for linux? We could probably leave this out if we want to just use whatever bpop sets as default.

**Review by Mrod1598 — CHANGES_REQUESTED** [review]

File: docs/configuration.md (hunk 5 above)
Comment (jsirianni) [reply docs/configuration.md:524]: Technically server is only officially supported on linux, and these configs are linux server ready.

**Review by Mrod1598 — APPROVED** [review]

**Pull request merged**